x-pack/plugins/apm/public/components/alerting/rule_types/register_apm_rule_types.ts

```diff
@@ -148,11 +149,29 @@ export function registerApmRuleTypes(
       return `${docLinks.links.alerting.apmRules}`;
     },
     ruleParamsExpression: lazy(() => import('./anomaly_rule_type')),
-    validate: () => ({
-      errors: [],
-    }),
+    validate: validateAnomalyRule,
     requiresAppContext: false,
     defaultActionMessage: anomalyMessage,
     priority: 90,
   });
 }
+
+function validateAnomalyRule(ruleParams: AlertParams) {
+  const validationResult = { errors: {} };
+  const errors: {
+    anomalyDetectorTypes?: string;
+  } = {};
+  validationResult.errors = errors;
+  if (
+    ruleParams.anomalyDetectorTypes &&
+    ruleParams.anomalyDetectorTypes.length < 1
+  ) {
+    errors.anomalyDetectorTypes = i18n.translate(
+      'xpack.apm.validateAnomalyRule.',
+      {
+        defaultMessage: 'At least one detector type is required',
+      }
+    );
+  }
+  return validationResult;
```

Comment:
Does this validation function require a string to be returned? Would be better if this just returned a boolean

Reply:
Please @sqren , take a look at my comments on the slack stand up, on why this needs to be like this, I tried already what are you suggesting

Reply:
Looks quite convoluted. What about this?

```ts
const anomalyDetectorTypeErrorMessage = i18n.translate(
  "xpack.apm.validateAnomalyRule.",
  {
    defaultMessage: "At least one detector type is required",
  }
);

return {
  errors: {
    anomalyDetectorTypes:
      ruleParams.anomalyDetectorTypes.length < 1
        ? anomalyDetectorTypeErrorMessage
        : undefined,
  },
};
```